lib/chef/provider/package.rb

```diff
@@ -207,6 +209,15 @@ def action_reconfig
       end
 
       # @todo use composition rather than inheritance
+
+      def with_subclass_array_wrapper(name, version)
+        if supports_arrays?
+          yield [name].flatten, [version].flatten
+        else
+          yield name, version
```

Comment:
Yeah, but we still can't tell yet.  If yum+apt are upgraded to us this (which they should) then we could think about that, but its still going to be a converge-time, not a compile-time raise since its in the provider.